integrations/watsonx/src/haystack_integrations/components/generators/watsonx/chat/chat_generator.py

```diff
@@ -0,0 +1,365 @@
+                    },
+                )
+            ],
+            "chunks": chunks,
+        }
+
+    def _handle_standard(self, api_args: dict[str, Any]) -> dict[str, Any]:
+        """Handle synchronous standard response."""
+        response = self.client.chat(messages=api_args["messages"], params=api_args["params"])
+        return self._process_response(response)
+
+    async def _handle_async_streaming(self, api_args: dict[str, Any]) -> dict[str, Any]:
+        """Handle asynchronous streaming response."""
+        chunks: list[StreamingChunk] = []
+        full_text = ""
+
+        stream = await self.client.achat_stream(messages=api_args["messages"], params=api_args["params"])
+
+        async for chunk in stream:
+            if not isinstance(chunk, dict) or not chunk.get("choices"):
+                continue
+
+            content = chunk["choices"][0].get("delta", {}).get("content", "")
+            if content:
+                full_text += content
+                chunk_meta = {
+                    "model": self.model,
+                    "index": chunk["choices"][0].get("index", 0),
+                    "finish_reason": chunk["choices"][0].get("finish_reason"),
+                    "received_at": datetime.now(timezone.utc).isoformat(),
+                }
+                chunks.append(StreamingChunk(content=content, meta=chunk_meta))
+
+        return {
+            "replies": [
+                ChatMessage.from_assistant(
+                    text=full_text,
+                    meta={
+                        "model": self.model,
+                        "finish_reason": chunks[-1].meta["finish_reason"] if chunks else "completed",
+                    },
+                )
+            ],
+            "chunks": chunks,
+        }
+
+    async def _handle_async_standard(self, api_args: dict[str, Any]) -> dict[str, Any]:
+        """Handle asynchronous standard response."""
+        response = await self.client.achat(**api_args)
+        return self._process_response(response)
+
+    def _process_response(self, response: dict[str, Any]) -> dict[str, Any]:
+        """Process standard response into Haystack format."""
+        if not response.get("choices"):
+            return {"replies": []}
+
+        choice = response["choices"][0]
+        message = choice.get("message", {})
+        content = message.get("content", "")
+        tool_calls = []
+
+        if "tool_calls" in message:
+            for tc in message["tool_calls"]:
+                if tc["type"] == "function":
+                    try:
+                        arguments = json.loads(tc["function"]["arguments"])
+                        tool_calls.append(
+                            ToolCall(id=tc.get("id"), tool_name=tc["function"]["name"], arguments=arguments)
+                        )
+                    except json.JSONDecodeError:
+                        logger.warning("Failed to parse tool call arguments: %s", tc["function"]["arguments"])
```

Comment:
Our haystack logger works a bit differently. Please format the string using this kind of syntax
```suggestion
                        logger.warning("Failed to parse tool call arguments: {tool_call_args}", tool_call_args=tc["function"]["arguments"])
```